### DISCUSSION TITLE
Registry Search API Issue

### DISCUSSION THREAD

**Discussion by shiva-rakshith** [Sunbird-RC/community#222] [Issues] [answered]
I have a property in my schema `roles` and the values can be `teacher`, `student`, `admin` etc. 
I am able to fetch the list of all teachers or students separately with the below search request.
```
{
    "filters": {
        "roles": {
            "contains": "teacher"
        }
    }
}
```
But I am not able to fetch the list if I give the combination of roles as below.
```
{
    "filters": {
        "roles": {
            "contains": ["teacher", "student"]
        }
    }
}
```
How the search filters should be if we want to fetch combinations of a property values?

cc: @vrayulu @maheshkumargangula

**Comment by tejash-jl** [comment, marked as answer]
@shiva-rakshith 
You need to use `or` operation for this.
```
{
    "filters": {
        "roles": {
            "or": ["teacher", "student"]
        }
    }
}
```

**Reply by shiva-rakshith** [reply]
@tejash-jl - Thanks for your response. The above suggested filters are working fine for fields of type `List`.
But for the fields of type `String`, it is not working. For example, I want to get two records, by giving osid's in filters as below. It is returning empty results.
```
{
    "filters": {
        "osid": {
            "or": ["5bd01f08-7f56-46ef-8452-7e6a45759304", "9f037ae0-5073-4188-873d-4aeaf4acdc25"]
        }
    }
}
```

**Reply by tejash-jl** [reply]
@shiva-rakshith this issue is fixed and pushed to the main branch. It'll be released shortly. You can clone/pull the main branch and run it locally for testing purposes.

**Reply by shiva-rakshith** [reply]
@tejash-jl Sure, Thank you!

**Comment by kuldeep-beehyv** [comment]
How do I get particular fields from the table?

**Reply by tejash-jl** [reply]
@kuldeep-beehyv,
you can do it via viewtemplates,
for example
```bash
curl --location --request POST 'http://localhost:8081/api/v1/Schema/search' \
--header 'Content-Type: application/json' \
--data-raw '{
    "filters": {
        
    },
    "viewTemplate": {
        "fields": [
            {
                "name": "name",
                "title": "fullName"
            }
        ]
    }
}'
```

here the `name` denotes the field name in the schema and `title` (optional) is used as the new field name

**Reply by kuldeep-beehyv** [reply]
Thank you @tejash-jl. Is there any documentation where all these are mentioned? I could not find any.
I need one more help, how do I use the not equals filter for a field. I observed there are eq, lt, gt. but didn't get any idea on the not equals filter
